### PR TITLE
Refactor LambdaExceptions into separate classes

### PR DESCRIPTION
**Pull request by romni**

### CODE REVIEW

File: src/main/java/io/leonis/zosma/exception/SupplierWithExceptions.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+   * A {@link Supplier} which may throw an exception.
+   *
+   * @return The object produced by the {@link Supplier}.
+   * @throws E The type of exception which may be thrown by the {@link Supplier}.
+   */
+  @SuppressWarnings("unchecked")
+  public S get() throws E {
+    try {
+      return function.get();
+    } catch (final Exception exception) {
```
Comment (thumbnail) [inline src/main/java/io/leonis/zosma/exception/SupplierWithExceptions.java]: can't you catch E here and be done with it ?

**Comment by thumbnail** [comment]
The guy that made this suggestion seems pretty smart

File: src/main/java/io/leonis/zosma/exception/RethrowExceptionBiConsumer.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * {@link Exception Exceptions} as {@link RuntimeException RuntimeExceptions}.
+ *
+ * @param <T> The type of the first argument passed to the {@link BiConsumerWithExceptions}.
+ * @param <U> The type of the second argument passed to the {@link BiConsumerWithExceptions}.
+ * @param <E> The type of exception which may be thrown by the {@link BiConsumerWithExceptions}.
+ * @author Rimon Oz
+ * @author Jeroen de Jong
+ */
+@Value
+public class RethrowExceptionBiConsumer<T, U, E extends Exception> {
```
Comment (romni) [inline src/main/java/io/leonis/zosma/exception/RethrowExceptionBiConsumer.java]: shouldn't this `extend BiConsumer<T, U>`?

Comment (thumbnail) [reply src/main/java/io/leonis/zosma/exception/RethrowExceptionBiConsumer.java]: That makes total sense. I'll add the extensions of the native types in a minute.

**Review by thumbnail — DISMISSED** [review]
Looks pretty good!

**Review by thumbnail — DISMISSED** [review]
Wow, such greatness

**Review by thumbnail — APPROVED** [review]

**Pull request merged**